astroquery/gaia/core.py

```diff
@@ -507,6 +735,18 @@ def __getCoordInput(self, value, msg):
         else:
             return value
 
+    @staticmethod
+    def correct_table_units(table):
```

Comment:
could you add a test for this with an example when the input table had the wrong units? Also, if e.g. the usage of `QTable` wouldn't solve this issue out of the box, and the issue is generic enough, we may want to upstream a solution for it into astropy.table